src/interpreter/Thread.as

```diff
@@ -111,8 +111,8 @@ public class Thread {
 	public function returnFromProcedure():Boolean {
 		for (var i:int = sp - 1; i >= 0; i--) {
 			if (stack[i].block.op == Specs.CALL) {
-				sp = i + 1;
-				popState();
+				sp = i + 1;  // 'hack' the stack pointer, but don't do the final popState here
+				block = null;  // make it do the final popState through the usual stepActiveThread mechanism
```

Comment:
Single line fix! :)
(The rest is just my attempt to explain what it means, so could be removed if you want...)